packages/strapi/lib/core/bootstrap.js

```diff
@@ -131,20 +109,15 @@ module.exports = function(strapi) {
       });
     });
 
-    Object.keys(plugin.models || []).forEach(key => {
-      let model = plugin.models[key];
-
-      Object.assign(model, {
-        __schema__: pickSchema(model),
-        modelType: 'contentType',
-        kind: getKind(model),
-        modelName: key,
-        uid: `plugins::${pluginName}.${key}`,
-        plugin: pluginName,
-        collectionName: model.collectionName || `${pluginName}_${key}`.toLowerCase(),
-        globalId: model.globalId || _.upperFirst(_.camelCase(`${pluginName}-${key}`)),
-        connection: model.connection || defaultConnection,
-      });
+    Object.keys(plugin.models || []).forEach(modelName => {
+      let model = plugin.models[modelName];
+
+      contentTypesUtils.formatContentType(
+        // mutate model
+        model,
+        { modelName, modelOrigin: 'plugin', defaultConnection },
```

Comment:
can we just check if there is a pluginName or an apiName instead of adding another param modelOrigin ?

Reply:
Yep, may lead to more misuses but it should be alright